service/matching/handler/engine.go

```diff
@@ -298,18 +303,19 @@ func (e *matchingEngineImpl) getTaskListByDomainLocked(domainID string, taskList
 }
 
 // For use in tests
-func (e *matchingEngineImpl) updateTaskList(taskList *tasklist.Identifier, mgr tasklist.Manager) {
+func (e *matchingEngineImpl) updateTaskList(taskList *tasklist.Identifier, mgr tasklist.ShardProcessor) {
 	e.taskListsLock.Lock()
 	defer e.taskListsLock.Unlock()
 	e.taskLists[*taskList] = mgr
 }
 
-func (e *matchingEngineImpl) removeTaskListManager(tlMgr tasklist.Manager) {
+func (e *matchingEngineImpl) removeTaskListManager(tlMgr tasklist.ShardProcessor) {
 	id := tlMgr.TaskListID()
 	e.taskListsLock.Lock()
 	defer e.taskListsLock.Unlock()
+
 	currentTlMgr, ok := e.taskLists[*id]
-	if ok && tlMgr == currentTlMgr {
+	if ok && currentTlMgr.String() == tlMgr.String() {
```

Comment:
Doing `.String` here is quite opaque. It's hard to tell what exactly is being compared without going to the implementation.